src/mgr.sol

```diff
@@ -110,38 +133,42 @@ contract TinlakeManager is LibNote {
 
     uint256 public constant dec = 18;
 
+    address public tranche;
+
     constructor(address vat_,      address dai_,
                 address daiJoin_,  address vow_,
                 address drop_,     address pool_,
-                address owner_,    address tranche,
-                bytes32 ilk_) public {
+                address operator_, address tranche_,
+                address end_,      bytes32 ilk_
+                ) public {
 
         vat = VatLike(vat_);
         dai = GemLike(dai_);
-        vow = VowLike(vow_);
-        daiJoin = GemJoinLike(daiJoin_);
+        end = EndLike(end_);
+        daiJoin = JoinLike(daiJoin_);
+        vow = vow_;
 
         gem = GemLike(drop_);
         pool = RedeemLike(pool_);
 
         ilk = ilk_;
         wards[msg.sender] = 1;
 
-        owner = owner_;
+        operator = operator_;
 
         safe = true;
         live = true;
         glad = true;
 
         dai.approve(daiJoin_, uint256(-1));
         vat.hope(daiJoin_);
-        gem.approve(pool_, uint256(-1));
-        gem.approve(tranche, uint256(-1));
+
+        tranche = tranche_;
```

Comment:
I know this comes from our comments, so actually directing this comment to @godsflaw and @brianmcmichael. Is it really worth it to save the variable to do partial approvals on `tell`? I mean if we can actually verify that `tranche` can only take `gem` when `tell` is called by governance we could avoid having to store this one.

Reply:
The tranche `redeemOrder` can only be triggered by Maker governance.  This is the normal way for DROP token holder to redeem their tokens. 
The mgr calls the `operator.redeemOrder`
redeemOrder in operator
https://github.com/centrifuge/tinlake/blob/develop/src/lender/operator.sol#L67
redeemOrder in tranche
https://github.com/centrifuge/tinlake/blob/develop/src/lender/tranche.sol#L128

I think we should just keep the `amount` approval then there is definitely no way that Tinlake
can access it.

In Tinlake we use a contract called `clerk` (operator in mgr terminology)  to interact with the `mgr`.
For all methods we require the following to be true:
```javascript
 function mkrActive() public view returns (bool) {
        return mgr.safe() && mgr.glad() && mgr.live();
    }
```
See: https://github.com/centrifuge/tinlake/blob/develop/src/lender/adapters/mkr/clerk.sol#L117